plugin/pkg/scheduler/schedulercache/cache.go

```diff
@@ -74,14 +74,23 @@ func newSchedulerCache(ttl, period time.Duration, stop <-chan struct{}) *schedul
 	}
 }
 
-func (cache *schedulerCache) GetNodeNameToInfoMap() (map[string]*NodeInfo, error) {
-	nodeNameToInfo := make(map[string]*NodeInfo)
+func (cache *schedulerCache) UpdateNodeNameToInfoMap(nodeNameToInfo map[string]*NodeInfo) error {
```

Comment:
This optimization seem to be enough. After this change, this no longer pops in profiles, so I would add more complications to this code (at least in the near term future)